File: nodejs/src/v3/delivery_metrics.js
import { ApiObject } from '../api_object.js';

// Use this class to get and to print all of the available
// advertising delivery metrics.
export class DeliveryMetrics extends ApiObject {
  // No constructor is required, because the parent constructor
  // should be called automagically. See:
  //   https://eslint.org/docs/rules/no-useless-constructor

  // https://developers.pinterest.com/docs/redoc/adtech_ads_v4/#operation/get_available_metrics_definition_handler
  // Get the full list of all available delivery metrics.
  // This call is not used much in day-to-day API code, but is a useful endpoint
  // for learning about the metrics.
  async get() {
    return (await this.request_data(
      '/ads/v4/resources/delivery_metrics')).metrics;
  }

  summary(delivery_metric) {
    return `${delivery_metric.name}: ${delivery_metric.definition}`;
  }

  // Print summary of a single metric.
  print(delivery_metric) {
    console.log(this.summary(delivery_metric));
  }

  // Print summary of data returned by get().
  print_all(delivery_metrics) {
    console.log('Available delivery metrics:');
    for (const [idx, metric] of delivery_metrics.entries()) {
      console.log(`[${idx + 1}] ${this.summary(metric)}`);
    }
  }
}

File: python/src/v3/delivery_metrics.py
from api_object import ApiObject


class DeliveryMetrics(ApiObject):
    """
    Use this class to get and to print all of the available
    advertising delivery metrics.
    """

    def __init__(self, api_config, access_token):
        super().__init__(api_config, access_token)

    def get(self):
        """
        https://developers.pinterest.com/docs/redoc/adtech_ads_v4/#operation/get_available_metrics_definition_handler
        Get the full list of all available delivery metrics.
        This call is not used much in day-to-day API code, but is a useful endpoint
        for learning about the metrics.
        """  # noqa: E501 because the long URL is okay
        return self.request_data("/ads/v4/resources/delivery_metrics").get("metrics")

    def summary(self, delivery_metric):
        return f"{delivery_metric['name']}: {delivery_metric['definition']}"

    def print(self, delivery_metric):
        """
        Print summary of a single metric.
        """
        print(self.summary(delivery_metric))

    def print_all(self, delivery_metrics):
        """
        Print summary of data returned by get().
        """
        print("Available delivery metrics:")
        for idx, metric in enumerate(delivery_metrics):
            print(f"[{idx + 1}] " + self.summary(metric))
